### PR TITLE
Fix block transform

### PR DESCRIPTION
**Pull request by ignasiialemany**
Use proper back and forth transforms to operate in true LOCAL

### CODE REVIEW

File: +MonteCarlo/@ParticleWalker/private/one_dt.m [hunk 1]
```diff
@@ -29,7 +29,8 @@
     end
 
     % substrate checks
-    [position_LOCAL, position_rotated, fn_RotReverse] = substrate.transformPosition(position(1, 1:3));
+    [position_LOCAL, fn_TransInverse, fn_Rot, fn_RotReverse] = substrate.transformPosition(position(1, 1:3));
```
Comment (janniklasrose) [inline +MonteCarlo/@ParticleWalker/private/one_dt.m]:
```suggestion
    [position_LOCAL, fn_TransformInverse, fn_Rot, fn_RotReverse] = substrate.transformPosition(position(1, 1:3));
```

**Review by janniklasrose — APPROVED** [review]
Looks good, happy to merge on successful test run

**Comment by ignasiialemany** [comment]
Testing the code it seems there is a typo in the `fn_TransformInverse` where the `shift` needs to be computed with `iZ+1`

File: +Substrate/@Transform/local2global.m [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+function [pos_global] = local2global(obj, pos_local, iX, iY, iZ)
 % transform a position inside block with indices iX, iY, iZ
 
+% Change block index to be zero-based
+[iX, iY, iZ] = deal(iX-1, iY-1, iZ-1);
+
 % translation offset
-offset = [iX+mod(iZ, 2)/2, iY, iZ].*obj.dxdydz_bb;
+if obj.shift_block
+    shift = mod(iZ, 2)/2;
```
Comment (ignasiialemany) [inline +Substrate/@Transform/local2global.m]:
```suggestion
    shift = mod(iZ+1, 2)/2;
```

**Comment by ignasiialemany** [comment]
Succesful test runs. Happy to merge!

**Pull request merged**